src/app/clusters/microwave-oven-control-server/microwave-oven-control-server.h

```diff
@@ -125,28 +131,50 @@ class Delegate
     virtual ~Delegate() = default;
 
     /**
-     *   @brief Handle Command Callback in application: SetCookingParameters
-     *   @return Returns the Interaction Model status code which was user determined in the business logic
-     *   @param  cookMode: the user defined modes which from the Microwave Oven Mode application level
-     *   @param  cookTime: the input cook time value
-     *   @param  powerSetting: the input power setting value
+     *   @brief Handle Command Callback in application: SetCookingParameters.
+     *   @return Returns the Interaction Model status code which was user determined in the business logic.
+     *   @param  cookMode: the user defined modes which from the Microwave Oven Mode application level.
+     *   @param  cookTime: the input cook time value.
+     *   @param  powerSetting: the input power setting value.
+     *   Note: if using power in watt, this parameter is expressed as WattSettingIndex.
```

Comment:
Seems like we should consider having a Variant or something here.

Failing that, the dependency on the feature map bits should be clearly documented.

Reply:
Both the type of PowerSetting and WattSettingIndex are uint8_t, Variant is not supported for the same type variables. 

Would using Optional<> is better?

Reply:
You could create types that wrap the uint8_t and use Variant...

But I have no strong opinions here.  I do think that the documentation needs to very clearly describe the setup, if it's not self-documenting.